homeassistant/components/rtsptowebrtc/config_flow.py

```diff
@@ -0,0 +1,36 @@
+"""Config flow for RTSPtoWebRTC."""
+from __future__ import annotations
+
+from typing import Any
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.data_entry_flow import FlowResult
+
+from . import DATA_SERVER_URL, DOMAIN
+
+
+class RTSPToWebRTCConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """RTSPtoWebRTC config flow."""
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Configure the RTSPtoWebRTC server url."""
+        if self._async_current_entries():
+            return self.async_abort(reason="single_instance_allowed")
```

Comment:
Any reason this single instance only? What if the user has two NVRs/two RTSPtoWebRTC servers ...or more?

Reply:
Happy to support this, but need to think through the details of how it would be used. Right now it's not clear to me how to use multiple.

Do you have thoughts about how to decide which instance to send the offer to at runtime?

Have a look at the code in __init__: it passes in an rtsp server url and does not use any of the existing configuration in the RTSPtoWebRTC server itself.

I think maybe what you are saying implies this integration could be pulling cameras configured in RTSPtoWebRTC and exposing them as home assistant cameras, which it's not doing now.  Right now this only does.the opposite and pushes cameras from home assistant to that server with zero extra configuration.

Am I understanding that right? I can see it makes sense to make it possible to support that.

Reply:
I'm probably over thinking this since my use case is probably on the extreme end.
  
Load balancing or high availability concerns don't really need to be considered here since we don't support it for Home Assistant anyways.

Reply:
I do you think you have a point if we added support for virtual cameras from RTSPtoWebRTC so I am open to considering this more.  Two thoughts:
- We could pick the "first" config entry for the existing live configuration use case
- Config entry unique ids:  Given the input is a url though that seems not allowed for a unique id:
https://developers.home-assistant.io/docs/config_entries_config_flow_handler#unacceptable-sources-for-a-unique-id -- is there a unique ID I could use now to better support this in the future?

Reply:
Does the server generate some type of UUID that persists across shutdown/startup?

Reply:
Nothing I am aware of, but i'll keep considering. Resolving for now.